imagebuildah/build.go

```diff
@@ -1250,12 +1256,22 @@ func getCreatedBy(node *parser.Node) string {
 // in the Dockerfile till the point of build we are at.
 // Used to verify whether a cache of the intermediate image exists and whether
 // to run the build again.
-func historyMatches(children []*parser.Node, history []v1.History) bool {
+func (b *Executor) historyMatches(children []*parser.Node, history []v1.History) bool {
 	i := len(history) - 1
 	for j := len(children) - 1; j >= 0; j-- {
 		instruction := children[j].Original
 		if children[j].Value == "run" {
 			instruction = instruction[4:]
+			buildArgs := b.getBuildArgs()
+			// If a previous image was built with some build-args but the new build process doesn't have any build-args
+			// specified, so compare the lengths of the old instruction with the current one
+			if buildArgs == "" && len(history[i].CreatedBy) > len(instruction)+11 {
```

Comment:
Might want to document and/or create a variable for the 11.

Reply:
Or, assuming --build-arg user=0, you're comparing the values of 
instruction: [echo $user]
history[i].CreatedBy: [user=0 /bin/sh -c echo $user]

Would it be better to just check and see if history[i].CreatedBy started with /bin/sh or not?

I' m on the fence.